packages/client/src/StreamVideoClient.ts

```diff
@@ -195,22 +195,22 @@ export class StreamVideoClient {
 
         // The call might already be tracked by the client,
         // if `call.created` was received before `call.ring`.
-        // In that case, we just reuse the already tracked call.
-        let theCall = this.writeableStateStore.findCall(call.type, call.id);
-        if (!theCall) {
-          // otherwise, we create a new call
-          theCall = new Call({
-            streamClient: this.streamClient,
-            type: call.type,
-            id: call.id,
-            members,
-            clientStore: this.writeableStateStore,
-            ringing: true,
-          });
-        }
-
+        // In that case, we cleanup the already tracked call.
+        const prevCall = this.writeableStateStore.findCall(call.type, call.id);
+        const prevMetadata = prevCall?.state.metadata;
+        await prevCall?.leave();
+        // we create a new call
+        const theCall = new Call({
+          streamClient: this.streamClient,
+          type: call.type,
+          id: call.id,
+          members,
+          clientStore: this.writeableStateStore,
+          ringing: true,
+          metadata: prevMetadata,
```

Comment:
the change is adding prevMetadata, so that auto drop do not happen due to change from undefined metadata timeout  to new metadata timeout